client/src/main/java/org/apache/rocketmq/client/trace/TraceBean.java

```diff
@@ -37,6 +37,15 @@ public class TraceBean {
     private String transactionId;
     private boolean fromTransactionCheck;
 
+    static {
+        byte[] ip = UtilAll.getIP();
+        if (UtilAll.isInternalIP(ip)) {
+            LOCAL_ADDRESS = UtilAll.ipToIPv4Str(ip);
+        } else {
+            LOCAL_ADDRESS = UtilAll.ipToIPv6Str(ip);
```

Comment:
Not being an internal IP does not necessarily mean it's an IPv6 IP.

Reply:
Thanks, it has been modified